# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.h
# Commit: 123b6c3a0ede
# Full Hash: 123b6c3a0ede48338825b52d0998818f5e76091c
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-23 21:43:09
# Regressor Bug: 1616775
# File Overlap Count: 1
# Description:
#   Bug 1616775 - P3. Add CompleteAllowAccessFor to support running in the parent. r=timhuang,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71012
# ==============================================================================

diff -r 414a605d677a -r 123b6c3a0ede toolkit/components/antitracking/ContentBlocking.h
--- a/toolkit/components/antitracking/ContentBlocking.h	Thu Apr 23 14:25:06 2020 +0000
+++ b/toolkit/components/antitracking/ContentBlocking.h	Thu Apr 23 14:28:41 2020 +0000
@@ -107,6 +107,16 @@
       const nsCString& aTrackingOrigin, int aAllowMode,
       uint64_t aExpirationTime =
           StaticPrefs::privacy_restrict3rdpartystorage_expiration());
+
+ private:
+  // This should be running either in the parent process or in the child
+  // processes with an in-process browsing context.
+  static MOZ_MUST_USE RefPtr<StorageAccessGrantPromise> CompleteAllowAccessFor(
+      dom::BrowsingContext* aParentContext, uint64_t aTopLevelWindowId,
+      nsIPrincipal* aTrackingPrincipal, const nsCString& aTrackingOrigin,
+      uint32_t aCookieBehavior,
+      ContentBlockingNotifier::StorageAccessGrantedReason aReason,
+      const PerformFinalChecks& aPerformFinalChecks = nullptr);
 };
 
 }  // namespace mozilla
